### PR TITLE
Use IL assemblies for NuGet packages to reduce size

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

The framework dependent build, builds IL assemblies. This PR uses those assemblies to build the packages. This drastically reduces the nuget package size.

## PR Context

We needed RID specific folders earlier due to crossgen. As crossgen'ed assemblies are runtime specific. Now that we are using IL assemblies, we just have `win` and `unix` runtimes as, IL assemblies work on all runtimes. We still need `win` and `unix` due to differences in source code due to conditional compilation.

Same as earlier, native dependencies are pulled in through references in the packages.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/hosting/NuGet.Config [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <configuration>
   <packageSources>
     <clear />
+    <add key="ilpkg" value="d:\temp\ilpackage" />
```
Comment (PaulHigin) [inline test/hosting/NuGet.Config]: I don't understand this change.  Where is "ilpkg" used, and how do we know this fixed path is valid?

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: test/hosting/NuGet.Config (hunk 1 above)
Comment (adityapatwardhan) [reply test/hosting/NuGet.Config]: Oops.. this should not be here. This was local testing.

Comment (adityapatwardhan) [reply test/hosting/NuGet.Config]: Reverted.

**Review by PaulHigin — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**